lib/ts/ink_align.h

```diff
@@ -25,67 +25,78 @@
 #ifndef _ink_align_h_
 #define _ink_align_h_
 
-#include "ts/ink_time.h"
-
-union Alias32 {
-  uint8_t byte[4];
-  int32_t i32;
-  uint32_t u32;
-};
-
-union Alias64 {
-  uint8_t byte[8];
-  int32_t i32[2];
-  uint32_t u32[2];
-  int64_t i64;
-  uint64_t u64;
-  ink_time_t i_time;
-};
+#include <memory>
+
+//
+// workalike to std::align() ... though it's not a great API
+//
+template <typename T_OBJ>
+inline T_OBJ *ats_align(std::size_t alignment, std::size_t size, T_OBJ *&ptr, std::size_t &space)
+{
+  intptr_t v = reinterpret_cast<intptr_t>(ptr);
+  ptrdiff_t vv = ((v-1) | (alignment-1)) + 1 - v;
+  if ( size + vv > space ) {
+    return nullptr;
+  }
+  space -= vv;
+  return (ptr = reinterpret_cast<T_OBJ*>(v + vv));
+}
 
 /**
  * Alignment macros
  */
 
 #define INK_MIN_ALIGN 8
 /* INK_ALIGN() is only to be used to align on a power of 2 boundary */
-#define INK_ALIGN(size, boundary) (((size) + ((boundary)-1)) & ~((boundary)-1))
+#define INK_ALIGN(size, boundary)  aligned_spacing(size,boundary)
```

Comment:
Okay ... not everywhere.  Whatever the case is ... I may put in another PR to lower the number of cut-pastes that have apparently been around a long time for alignments.